src/main/java/internet/market/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,199 @@
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add shopping cart with id = "
+                    + item.getId(), e);
+        }
+        addProductsToShoppingCart(item.getProducts(), item.getId());
+        return item;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts WHERE "
+                + "deleted = false AND shopping_cart_id = ?";
+        ShoppingCart shoppingCart = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = getShoppingCartInformationFromDataBase(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get shoppingCart with id = " + id, e);
+        }
+        if (shoppingCart != null) {
+            shoppingCart.setProducts(getProductsFromCart(id));
+        }
+        return Optional.ofNullable(shoppingCart);
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        List<ShoppingCart> allCarts = new ArrayList<>();
+        String query = "SELECT * FROM shopping_carts WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                allCarts.add(getShoppingCartInformationFromDataBase(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all shoppingCarts", e);
+        }
+        for (ShoppingCart cart : allCarts) {
+            cart.setProducts(getProductsFromCart(cart.getId()));
+        }
+        return allCarts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart item) {
+        String query = "UPDATE shopping_carts SET user_id = ? "
+                + "WHERE shopping_cart_id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, item.getUserId());
+            statement.setLong(2, item.getId());
+            statement.executeUpdate();
+            List<Product> allProducts = getProductsFromCart(item.getId());
+            deleteProductsFromCart(item.getId());
+            addProductsToShoppingCart(allProducts, item.getId());
```

Comment:
both these two methods open connections inside them
let's avoid the situation where we have a connection inside another connection

Reply:
done